sdk/synapse/Azure.Analytics.Synapse.Spark/src/Customization/SparkBatchOperation.cs

```diff
@@ -0,0 +1,157 @@
+
+namespace Azure.Analytics.Synapse.Spark
+{
+    /// <summary>
+    /// A long-running operation for:
+    /// <see cref="SparkBatchClient.StartCreateSparkBatchJobAsync(SparkBatchJobOptions, bool?, CancellationToken)"/>,
+    /// <see cref="SparkBatchClient.StartCreateSparkBatchJob(SparkBatchJobOptions, bool?, CancellationToken)"/>,
+    /// <see cref="SparkBatchClient.StartGetSparkBatchJobAsync(int, bool?, CancellationToken)"/>,
+    /// <see cref="SparkBatchClient.StartGetSparkBatchJob(int, bool?, CancellationToken)"/>
+    /// </summary>
+    public class SparkBatchOperation : Operation<SparkBatchJob>
+    {
+        private static readonly TimeSpan s_defaultPollingInterval = TimeSpan.FromSeconds(5);
+
+        private readonly ClientDiagnostics _diagnostics;
+
+        private readonly SparkBatchClient _client;
+        private readonly SparkBatchJob _value;
+        private Response<SparkBatchJob> _response;
+        private bool _completed;
+        private RequestFailedException _requestFailedException;
+
+        private static HashSet<string> SessionSubmissionFinalStates = new HashSet<string>
+        {
+            "error",
+            "dead",
+            "success",
+            "killed"
+        };
+
+        internal SparkBatchOperation(SparkBatchClient client, ClientDiagnostics diagnostics, Response<SparkBatchJob> response)
+        {
+            _client = client;
+            _value = response.Value ?? throw new InvalidOperationException("The response does not contain a value.");
+            _response = response;
+            _diagnostics = diagnostics;
+        }
+
+        /// <inheritdoc/>
+        public override string Id => _value.Id.ToString(CultureInfo.InvariantCulture);
+
+        /// <summary>
+        /// Gets the <see cref="SparkBatchJob"/>.
+        /// You should await <see cref="WaitForCompletionAsync(CancellationToken)"/> before attempting to use session in this pending state.
+        /// </summary>
+        /// <remarks>
+        /// Azure Synapse will return a <see cref="SparkBatchJob"/> immediately but may take time to the session to be ready.
+        /// </remarks>
+        public override SparkBatchJob Value
+        {
+            get
+            {
+#pragma warning disable CA1065 // Do not raise exceptions in unexpected locations
+                if (!HasCompleted)
+                {
+                    throw new InvalidOperationException("The operation is not complete.");
+                }
+                if (_requestFailedException != null)
+                {
+                    throw _requestFailedException;
+                }
+#pragma warning restore CA1065 // Do not raise exceptions in unexpected locations
+                return _value;
+            }
+        }
+
+        /// <inheritdoc/>
+        public override bool HasCompleted => _completed;
+
+        /// <inheritdoc/>
+        public override bool HasValue => !ResponseHasError && HasCompleted;
```

Comment:
would it be the same to use `_requestFailedException != null` here instead of creating a new variable?

Reply:
It's slightly different:
- _requestFailedException means that we ended in livy state error OR we had some exception such as a request failed. In theory those can be transient (if networking or service fixes self).
- _responseHasError means we ended up in the livy state of error, which is a final state.